mypyc/primitives/struct_regsitry.py

```diff
@@ -1,24 +1,30 @@
 """Struct registries for C backend"""
 
-from typing import List, NamedTuple
-from mypyc.ir.rtypes import RType
+from typing import List, Dict
+from mypyc.ir.rtypes import RType, StructInfo, RStruct, c_pyssize_t_rprimitive, pointer_rprimitive
 
-CStructDescription = NamedTuple(
-    'CStructDescription', [('name', str),
-                           ('names', List[str]),
-                           ('types', List[RType])])
+struct_infos = {}  # type: Dict[str, StructInfo]
+struct_types = {}  # type: Dict[str, RStruct]
 
 
 def c_struct(name: str,
              names: List[str],
-             types: List[RType]) -> CStructDescription:
+             types: List[RType]) -> StructInfo:
     """Define a known C struct for generating IR to manipulate it
 
     name: The name of the C struct
     types: type of each field
     names: name of each field
            TODO: the names list can be empty in the future when we merge Tuple as part of Struct
     """
-    return CStructDescription(name, names, types)
+    info = StructInfo(name, names, types)
+    struct_infos[name] = info
+    typ = RStruct(info)
+    struct_types[name] = typ
+    return info
 
-# TODO: create PyVarObject, to do which we probably need PyObject
+
+c_struct(
+    name='PyVarObject',
+    names=['ob_refcnt', 'ob_type', 'ob_size'],
+    types=[c_pyssize_t_rprimitive, pointer_rprimitive, c_pyssize_t_rprimitive])
```

Comment:
`PyVarObject` is defined like this in Python 3.7:

```c
typedef struct {
    PyObject ob_base;
    Py_ssize_t ob_size; /* Number of items in variable part */
} PyVarObject;
```

For consistency, we should define the `PyObject` struct and use `PyObject` as part of the definition of `PyVarObject` struct, instead of expanding the fields.